### PR TITLE
added fip_to_underlay example

### PR DESCRIPTION
**Pull request by networkop**
Hi,
i've added a short example for FIP to VSG/VRS-G provisioning. Tested with VSR-G and seems to be working fine.

### CODE REVIEW

File: docs/fip_to_gateway.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  --vlan VRS_VLAN       VRS Gateway Network Interface Vlan ID
+
+### Example ###
+python fip_to_gateway.py -E csp -H 10.6.132.43 -p csproot -u csproot -S \
+       --address 10.6.132.160 --mask 255.255.255.224 --gw 10.6.132.190 \
+       --ip 10.6.132.161 --mac f4:cc:55:e0:14:00 --fip 10.6.132.192 \
+       --vrs 10.6.132.47 --port eth0 --vlan 205
+
+### Requirements ###
+* Nuage VSPK/VSDK (3.2+)
```
Comment (pdellaert) [inline docs/fip_to_gateway.md]: As your code looks for v4_0 specifically, this should state 'Nuage VSPK/VSDK 4.0'

File: fip_to_gateway.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    nuage_user = nc.user
+
+    logger.debug('Getting FIP subnet parent ID')
+    fip_obj = nuage_user.subnets.get_first(filter="address == '{0}'".format(fip_net))
+    shared_resource_id = fip_obj.parent_id
+    logger.debug('FIP parent ID is: {0}'.format(shared_resource_id))
+
+    logger.debug('Creating a new VLAN on Gateway port')
+    new_vlan = vsdk.NUVLAN(value=vrs_vlan)
+    gw = nuage_user.gateways.get_first(filter="name == '{0}'".format(vrs_ip))
```
Comment (pdellaert) [inline fip_to_gateway.py]: Gateway names can be changed, they do not necessarily remain the IP of the VRS-G/VRS... Somehow this needs to be clear to users. 

I would suggest to change the help text of the ```--vrs``` parameter to 
`The name of the Gateway as it appears in your infrastructure (defaults to IP if you have not changed it)`

For clearness sake i would also change the variable name to ```vrs_name```

Also add a check to see if you actually found a matching gw before continuing. If not, exit with an error and an error return code.

File: fip_to_gateway.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    except Exception, e:
+        logger.error('Could not connect to Nuage host %s with user %s and specified password' % (nuage_host, nuage_username))
+        logger.critical('Caught exception: %s' % str(e))
+        return 1
+
+    nuage_user = nc.user
+
+    logger.debug('Getting FIP subnet parent ID')
+    fip_obj = nuage_user.subnets.get_first(filter="address == '{0}'".format(fip_net))
+    shared_resource_id = fip_obj.parent_id
```
Comment (pdellaert) [inline fip_to_gateway.py]: Add a check to see if you actually found the FIP object, if not, return an error and exit with an error return code.

File: fip_to_gateway.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    logger.debug('FIP parent ID is: {0}'.format(shared_resource_id))
+
+    logger.debug('Creating a new VLAN on Gateway port')
+    new_vlan = vsdk.NUVLAN(value=vrs_vlan)
+    gw = nuage_user.gateways.get_first(filter="name == '{0}'".format(vrs_ip))
+    port = gw.ports.get_first(filter="name == '{0}'".format(vrs_port))
```
Comment (pdellaert) [inline fip_to_gateway.py]: Add a check to see if you actually found a matching port before continuing. If not, exit with an error and an error return code.

**Comment by networkop** [comment]
Hi Philippe,
all done. I've added some additional fault checking for port and vlan to make sure it's created on the right type of port, a part of the pre-configured range and doesn't exist before it's been created. Let me know what you think

**Comment by pdellaert** [comment]
Looks great! Thanks for the addition, changes and support!

**Comment by networkop** [comment]
No worries, pleasure to help.

**Pull request merged**